process/coordinator/process.go

```diff
@@ -325,39 +334,27 @@ func (tc *transactionCoordinator) RemoveBlockDataFromPool(body block.Body) error
 }
 
 // ProcessBlockTransaction processes transactions and updates state tries
-func (tc *transactionCoordinator) ProcessBlockTransaction(body block.Body, round uint64, haveTime func() time.Duration) error {
+func (tc *transactionCoordinator) ProcessBlockTransaction(
+	body block.Body,
+	round uint64,
+	haveTime func() time.Duration,
+) error {
 	separatedBodies := tc.separateBodyByType(body)
 
-	var errFound error
-	errMutex := sync.Mutex{}
-
-	// TODO: think if it is good in parallel or it is needed in sequences
-	wg := sync.WaitGroup{}
-	wg.Add(len(separatedBodies))
-
-	for key, value := range separatedBodies {
-		go func(blockType block.Type, blockBody block.Body) {
-			preproc := tc.getPreprocessor(blockType)
-			if preproc == nil {
-				wg.Done()
-				return
-			}
-
-			err := preproc.ProcessBlockTransactions(blockBody, round, haveTime)
-			if err != nil {
-				log.Debug(err.Error())
+	// processing has to be done in order, as the order of different type of transactions over the same account is strict
+	for blockType, blockBody := range separatedBodies {
```

Comment:
This is still wrong as separatedBodies is a map. Iterating through it might lead different results. Make use of the sorted keys.

Reply:
agreed. changed.